src/control-plane/backend/lambda/api/test/api/auth.test.ts

```diff
@@ -109,49 +109,82 @@ describe('Validate role middleware test', () => {
+    dictionaryMock(ddbMock);
     userMock(ddbMock, 'fake@example.com', IUserRole.ADMIN, false);
     const res = await request(app)
-      .get('/api/user/details?uid=fake@example.com')
+      .get('/api/user/details?id=fake@example.com')
       .set(amznRequestContextHeader, context);
     expect(res.statusCode).toBe(200);
     expect(res.body.data.role).toEqual(IUserRole.OPERATOR);
-    expect(ddbMock).toHaveReceivedCommandTimes(GetCommand, 2);
-    expect(ddbMock).toHaveReceivedCommandTimes(PutCommand, 1);
+    expect(ddbMock).toHaveReceivedCommandTimes(GetCommand, 4);
+    expect(ddbMock).toHaveReceivedCommandTimes(PutCommand, 0);
     expect(ddbMock).toHaveReceivedCommandTimes(UpdateCommand, 0);
   });
 
   it('User not in DDB and error group in token.', async () => {
+    dictionaryMock(ddbMock);
     userMock(ddbMock, 'fake@example.com', IUserRole.ADMIN, false);
     const res = await request(app)
-      .get('/api/user/details?uid=fake@example.com')
+      .get('/api/user/details?id=fake@example.com')
       .set(amznRequestContextHeader, context_error_group);
     expect(res.statusCode).toBe(200);
     expect(res.body.data.role).toEqual(IUserRole.NO_IDENTITY);
+    expect(ddbMock).toHaveReceivedCommandTimes(GetCommand, 4);
+    expect(ddbMock).toHaveReceivedCommandTimes(PutCommand, 0);
+    expect(ddbMock).toHaveReceivedCommandTimes(UpdateCommand, 0);
+  });
+
+  it('Get User settings with current user not in DDB and error group in token.', async () => {
+    dictionaryMock(ddbMock);
+    userMock(ddbMock, 'fake@example.com', IUserRole.ADMIN, false);
+    const res = await request(app)
+      .get('/api/user/settings')
+      .set(amznRequestContextHeader, context_error_group);
+    expect(res.statusCode).toBe(403);
+    expect(res.body).toEqual({ message: 'Insufficient permissions to access the API.', success: false });
+    expect(ddbMock).toHaveReceivedCommandTimes(GetCommand, 2);
+    expect(ddbMock).toHaveReceivedCommandTimes(PutCommand, 0);
+    expect(ddbMock).toHaveReceivedCommandTimes(UpdateCommand, 0);
+  });
+
+  it('Get User settings with current user in DDB and error group in token.', async () => {
+    dictionaryMock(ddbMock);
+    userMock(ddbMock, 'fake@example.com', IUserRole.ADMIN, true);
+    const res = await request(app)
+      .get('/api/user/settings')
+      .set(amznRequestContextHeader, context_error_group);
+    expect(res.statusCode).toBe(200);
+    expect(res.body.data).toEqual({
+      analystRoleNames: 'ClickstreamAnalyst',
+      operatorRoleNames: 'ClickstreamOperator',
+      roleJsonPath: '$.payload.cognito:groups',
+    });
     expect(ddbMock).toHaveReceivedCommandTimes(GetCommand, 2);
-    expect(ddbMock).toHaveReceivedCommandTimes(PutCommand, 1);
+    expect(ddbMock).toHaveReceivedCommandTimes(PutCommand, 0);
     expect(ddbMock).toHaveReceivedCommandTimes(UpdateCommand, 0);
   });
 
   it('User in DDB and no group in token.', async () => {
```

Comment:
Let's add a case for,

- the user is an analyst in group mapping
- the user is given as an operator in user setting

What is the expected final user role?

Reply:
add case: `User role is analyst in DDB and operator role map from token.` 
expected: analyst.